src/utils/LocalDiscoveryServer.ts

```diff
@@ -2,8 +2,15 @@ import WebSocket, { WebSocketServer, AddressInfo } from "ws";
 import P2PManager from "@/P2PManager";
 import { LocalTransport } from "@/transport";
 import { ChannelId } from "@/types/types";
+// Import directly to avoid circular dependency through the utils barrel
+import { createLogger, Logger } from "@/utils/PeerLogger";
 
 const MAX_PORT_RETRIES = 20;
+const LOCAL_WS_HOST = "127.0.0.1";
+const WS_CONNECT_TIMEOUT_MS = 750;
+const REGISTRY_CONNECT_MAX_RETRIES = 10;
+
+type DiscoveryMode = "registry" | "peer";
```

Comment:
i'm not strictly against, but somewhat overenigneerd for sole purpose of logging
you could just as well write the correct string inline in the log